### PR TITLE
Performance tool: fixes to run across branches

### PR DESCRIPTION
**Pull request by youknowriad**

### CODE REVIEW

File: bin/plugin/commands/performance.js [hunk 1]
```diff
@@ -122,12 +122,14 @@ async function getPerformanceResultsForBranch(
 	environmentDirectory,
 	branch
 ) {
+	await git.discardLocalChanges( environmentDirectory );
```
Comment (youknowriad) [inline bin/plugin/commands/performance.js:125]: This ensures that if there's a package-lock.diff lingering it's discarded.

File: bin/plugin/commands/performance.js [hunk 2]
```diff
@@ -122,12 +122,14 @@ async function getPerformanceResultsForBranch(
...
+
 	log( '>> Fetching the ' + formats.success( branch ) + ' branch' );
 	await git.checkoutRemoteBranch( environmentDirectory, branch );
 
 	log( '>> Building the ' + formats.success( branch ) + ' branch' );
 	await runShellScript(
-		'npm install && npm run build',
+		'rm -rf node_modules && npm install && npm run build',
```
Comment (youknowriad) [inline bin/plugin/commands/performance.js:132]: Switching branches and npm install sometimes cause issues unless you remove the node modules. This is a bit less performant but more stable.

File: bin/plugin/commands/performance.js [hunk 3]
```diff
@@ -205,7 +207,7 @@ async function runPerformanceTests( branches ) {
 	);
 
 	log( '>> Starting the WordPress environment' );
-	await runShellScript( 'npm run wp-env start' );
+	await runShellScript( 'npm run wp-env start', environmentDirectory );
```
Comment (youknowriad) [inline bin/plugin/commands/performance.js:210]: This is the main issue we had, the environment was not updating when switching branches on the temporary clone

File: packages/e2e-test-utils/src/inserter.js [hunk 4]
```diff
@@ -3,6 +3,11 @@
  */
 import { pressKeyWithModifier } from './press-key-with-modifier';
 
+// This selector is written to support the current and old inserter markup
+// because the performance tests need to be able to run across versions.
+const INSERTER_SEARCH_SELECTOR =
```
Comment (youknowriad) [inline packages/e2e-test-utils/src/inserter.js:8]: This ensures the tests run on wp/5.4 branch too. We need to support two major releases for the performance tests in order to produce the recap on each plugin release post.

File: bin/plugin/commands/performance.js (hunk 1 above)
Comment (aduth) [reply bin/plugin/commands/performance.js:125]: > This ensures that if there's a package-lock.diff lingering it's discarded.

Could make sense as an inline comment.

```suggestion
	// Restore clean working directory (e.g. if `package-lock.json` has local
	// changes after install).
	await git.discardLocalChanges( environmentDirectory );
```

File: bin/plugin/commands/performance.js (hunk 2 above)
Comment (aduth) [reply bin/plugin/commands/performance.js:132]: > Switching branches and npm install sometimes cause issues unless you remove the node modules. This is a bit less performant but more stable.

Do you think there could be a similar issue with lingering build artifacts? The ones that would be removed by `npm run clean:packages` and `npm run clean:package-types`.

**Review by aduth — APPROVED** [review]

File: bin/plugin/commands/performance.js (hunk 2 above)
Comment (youknowriad) [reply bin/plugin/commands/performance.js:134]: I  thought `npm  run build` would take care of that?

Comment (youknowriad) [reply bin/plugin/commands/performance.js:134]: It seems the package cleaning is indeed performed but not the types one. I think it's not impactful for the performance tests but we maybe want to run the type cleaning as a "pre" command similar to the build cleaning.

Comment (youknowriad) [reply bin/plugin/commands/performance.js:134]: Mmm, yeah maybe. Worth a try

**Pull request merged**